templates/deployments/appname/app.jsonnet.tpl

```diff
@@ -369,6 +389,40 @@ local all = {
 			},
 		},
 	},
+  {{- if (stencil.Arg "hpa.enabled") }}
+    hpa: ok.HorizontalPodAutoscaler(app.name, app.namespace) {
+      apiVersion: 'autoscaling/v2beta2',
+      target:: $.deployment,
+      spec+: if (isDev) then {
+        minReplicas: currentHpaReplicaConfig.minReplicas,
+        maxReplicas: currentHpaReplicaConfig.maxReplicas,
+        behavior+: {
+          {{- if (stencil.Arg "hpa.scaleDown.stabilizationWindowSeconds") }}
+          scaleDown: {
+            stabilizationWindowSeconds: {{ stencil.Arg "hpa.scaleDown.stabilizationWindowSeconds" }},
+          },
+          {{- end }}
+          {{- if (stencil.Arg "hpa.scaleUp.stabilizationWindowSeconds") }}
+          scaleUp: {
+            stabilizationWindowSeconds: {{ stencil.Arg "hpa.scaleUp.stabilizationWindowSeconds" }},
+          },
+          {{- end }}
+        },
+        {{- if (stencil.Arg "hpa.cpuUtilization") }}
+        metrics+: [{
```

Comment:
So I think one of the issues I have with all of these blocks is that it gets complicated, do I override it in a block or the overrides? I was trying to by pass this completely by simply only writing what is specified, if people want to override they can.

Though I can add blocks I guess. I'm just worried about adding "metrics" and having it empty and it not working out of the box

Reply:
Whenever I do something like you suggested above I end up regretting it. After its produced there is now no control over it. I have seen so many different ways to override settings so there is a bit of confusion on my side which is the best way to go.

Reply:
@jkinkead Here are the things I want to avoid doing
1) Adding yet another file that people have to use to configure hpa configuration alone.
2) I dislike adding blocks to the app.jsonnet as I find them to just get in the way in the future.

I would like to do the following
1) Keep what I'm doing and have it ONLY add the configuration if its set. 
2) Add documentation to this PR and add it to the description of the HPA enabled parameters so that IF people want to override they can copy and paste jsonnet into the app.config or the overrides.config files to get exactly the configuration you want.

I think I just want to enable the happy path, make that SUPER easy, then if people want something more specific, they can grab a snippet and just drop in an override.